lib/src/components/NxTabs/__tests__/NxTabs.test.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import { mount, shallow } from 'enzyme';
+import NxTabs from '../NxTabs';
+import NxTabList from '../../NxTabList/NxTabList';
+import NxTab from '../../NxTab/NxTab';
+import NxTabPanel from '../../NxTabPanel/NxTabPanel';
+
+describe('NxTabs', function () {
+  it('renders with a unique id each time', function () {
+    const firstTabs = shallow(
+      <NxTabs activeTab={0}>
+        <NxTabList>
+          <NxTab>Tabs 0 Tab 0</NxTab>
+        </NxTabList>
+      </NxTabs>
+    );
+    const secondTabs = shallow(
+      <NxTabs activeTab={0}>
+        <NxTabList>
+          <NxTab>Tabs 1 Tab 0</NxTab>
+        </NxTabList>
+      </NxTabs>
+    );
+
+    expect(firstTabs).not.toHaveProp('id', secondTabs.props().id);
+  });
+
+  it('activates the active tab', function () {
+    const component = mount(
+      <NxTabs activeTab={1}>
+        <NxTabList>
+          <NxTab>Tab 0</NxTab>
+          <NxTab>Tab 1</NxTab>
+        </NxTabList>
+        <NxTabPanel>Content 0</NxTabPanel>
+        <NxTabPanel>Content 1</NxTabPanel>
+      </NxTabs>
+    );
+
+    expect(component.find('[role="tab"]').first()).not.toHaveClassName('active');
+    expect(component.find('[role="tab"]').last()).toHaveClassName('active');
+    expect(component.find('[role="tabpanel"]')).toHaveText('Content 1');
+  });
+
+  it('calls onTabSelect when a tab is clicked', function() {
+    const onTabSelect = jest.fn();
+    const component = mount(
+      <NxTabs activeTab={0} onTabSelect={onTabSelect}>
+        <NxTabList>
+          <NxTab>Tab 0</NxTab>
+        </NxTabList>
+      </NxTabs>
+    );
+
+    component.find('[role="tab"]').simulate('click');
+
+    expect(onTabSelect).toHaveBeenCalledWith(0);
+  });
+
+  it('calls onTabSelect when a tab is selected via keyboard', function () {
+    const onTabSelect = jest.fn();
+    const component = mount(
+      <NxTabs activeTab={0} onTabSelect={onTabSelect}>
+        <NxTabList>
+          <NxTab>Tab 0</NxTab>
+        </NxTabList>
+      </NxTabs>
+    );
+
+    component.find('[role="tab"]').simulate('keypress', { key: ' ' });
+
+    expect(onTabSelect).toHaveBeenCalledWith(0);
+  });
+});
```

Comment:
Ideally would also have the following tests:

* when `activeTab` is unspecified
* That the `nx-tabs` classname is applied
* when the children validity checks fail (e.g. when the first child is not a valid element or does not have valid children)
* That child ids are working correctly (I recommend shallow rendering and just checking that the rootId prop is being passed correctly for that)

Reply:
b26b775

Reply:
> b26b775

Still missing the case when `activeTab` is unspecified.  It could be added to the `renders no tab contents when none are active` test as it is a variant of that.

Reply:
fcfd721